### PR TITLE
`Submission`: Prevent string submissions over max length

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Short answers in quiz exercises could be too long to be stored in the database (no client or server safeguards).

### Description

This PR adds a client and server check for the length of short answers in quizzes.
Additionally, the validation for modeling, text and quiz exercises were unified using the `@Size` and `@Valid` annotation.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Modeling, Text and Quiz exercise
- 1 Exam with a Modeling, Text and Quiz exercise

1. Log in to Artemis
2. Navigate to Course Exercises
3. Participate in the exercises and try to submit text over the limit
4. Repeat the test with an exam

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

Added server tests for modeling, text and quiz exercise submission size.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/config/Constants.java [hunk 1]
```diff
@@ -97,9 +97,11 @@ public final class Constants {
     public static final long MAX_SUBMISSION_FILE_SIZE = 8 * 1024 * 1024; // 8 MB
 
     // Note: The values in input.constants.ts (client) need to be the same
-    public static final long MAX_SUBMISSION_TEXT_LENGTH = 30 * 1000; // 30.000 characters
+    public static final int MAX_SUBMISSION_TEXT_LENGTH = 30 * 1000; // 30.000 characters
 
-    public static final long MAX_SUBMISSION_MODEL_LENGTH = 100 * 1000; // 100.000 characters
+    public static final int MAX_SUBMISSION_MODEL_LENGTH = 100 * 1000; // 100.000 characters
+
+    public static final int MAX_QUIZ_ANSWER_TEXT_LENGTH = 1000; // 1.000 characters
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/config/Constants.java:104]:
```suggestion
    public static final int MAX_SHORT_ANSWER_QUIZ_TEXT_LENGTH = 255;
```

File: src/main/java/de/tum/in/www1/artemis/domain/modeling/ModelingSubmission.java [hunk 2]
```diff
@@ -23,10 +27,12 @@
 public class ModelingSubmission extends Submission {
 
     @Column(name = "model")
+    @Size(max = MAX_SUBMISSION_MODEL_LENGTH, message = "The modeling submission is too large.")
     @Lob
     private String model;
 
     @Column(name = "explanation_text")
+    @Size(max = MAX_SUBMISSION_TEXT_LENGTH)
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/domain/modeling/ModelingSubmission.java:35]:
```suggestion
    @Size(max = MAX_SUBMISSION_TEXT_LENGTH, message = "The explanation of the modeling submission is too large.")
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JohannesStoehr — DISMISSED** [review]
Code looks good and should cover the error.
We should use @Valid more often in the future 💪

**Review by JohannesStoehr — APPROVED** [review]
Re-approve

**Review by stefanwaldhauser — APPROVED** [review]
Good change!

**Review by manuelmanso — APPROVED** [review]
Tested on ts3, works as expected. Code LGTM, also verified that the `maxCharacterCount` wasn't missing in any of the inputs in the client. Nice use of the `assertThatDb` too 👍

**Review by jpbernius — APPROVED** [review]

**Pull request merged**